File: backend/app/schemas/student.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr


class StudentBase(BaseModel):
    student_id: str
    first_name: str
    last_name: str
    preferred_name: Optional[str] = None
    email: Optional[str] = None
    grade: Optional[int] = None
    division: Optional[str] = None
    graduation_year: Optional[int] = None
    is_active: bool = True


class StudentCreate(StudentBase):
    pass


class StudentUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    preferred_name: Optional[str] = None
    email: Optional[str] = None
    grade: Optional[int] = None
    division: Optional[str] = None
    graduation_year: Optional[int] = None
    is_active: Optional[bool] = None


class StudentRead(StudentBase):
    id: int
    ad_object_id: Optional[str] = None
    canvas_user_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class StudentSummary(BaseModel):
    """Lightweight student record for list views."""
    id: int
    student_id: str
    first_name: str
    last_name: str
    preferred_name: Optional[str] = None
    grade: Optional[int] = None
    division: Optional[str] = None
    graduation_year: Optional[int] = None
    is_active: bool

    model_config = {"from_attributes": True}

File: frontend/src/types/Student.ts

export type Division = "LS" | "MS" | "HS";
 
export interface Student {
  id: number;
  student_id: string;
  first_name: string;
  last_name: string;
  preferred_name: string | null;
  email: string | null;
  grade: number | null;
  division: Division | null;
  graduation_year: number | null;
  is_active: boolean;
}
 
export const DIVISIONS: Division[] = ["LS", "MS", "HS"];
 
export const DIVISION_LABELS: Record<Division, string> = {
  LS: "Lower School",
  MS: "Middle School",
  HS: "High School",
};
 
export const GRADES_BY_DIVISION: Record<Division, number[]> = {
  LS: [1, 2, 3, 4, 5],
  MS: [6, 7, 8],
  HS: [9, 10, 11, 12],
};
 